pg_search/src/postgres/mod.rs

```diff
@@ -154,6 +154,15 @@ impl ParallelScanPayload {
             let segments_slice: &mut [[u8; SEGMENT_INFO_SIZE]] =
                 std::slice::from_raw_parts_mut(ptr.cast(), segments.len());
 
+            // resort the segments, smallest to largest by document count
+            //
+            // when segments are claimed by workers they're claimed from back-to-front
+            // and our goal is to have the largest segments claimed first so that
+            // the processing done on them takes longer, allowing more workers to
+            // checkout their own segments
```

Comment:
It seems reasonable not to have a test for this, but maybe we could we put a note on `checkout_segment` referencing this heuristic, to try and ensure that it doesn't break during a refactoring?